### PR TITLE
return constant if possible + don't compute shouldUpdateBranch unless necessary

### PR DESCRIPTION
**Pull request by adhusson**
Very small optim, more for code clarity than anything else, but still must be checked

### CODE REVIEW

File: src/MgvHasOffers.sol [hunk 1]
```diff
@@ -97,8 +92,10 @@ contract MgvHasOffers is MgvCommon {
           nextOfferData.offer = nextOfferData.offer.prev(prevId);
         }
         if (prevId != 0 && nextId != 0) {
-          return (local, shouldUpdateBranch);
+          return (local, false);
         }
+
+        shouldUpdateBranch = shouldUpdateBranch && prevId == 0 && !bestBin.strictlyBetter(offerBin);
```
Comment (espendk) [inline src/MgvHasOffers.sol:98]: I don't understand why we should update the branch if `bestBin == offerBin`? Doesn't that mean that `local` already has everything cached correctly?

(I know this wasn't changed here, but it struck me when reading the code)

File: src/MgvHasOffers.sol [hunk 2]
```diff
@@ -74,11 +74,6 @@ contract MgvHasOffers is MgvCommon {
         // with
         // offerId == local.best() (but best will maybe go away in the future)
 
-        // If shouldUpdateBranch is false is means we are about to insert anyway, so no need to load the best branch right now
-        // if local.bin < offerBin then a better branch is already cached. note that local.bin >= offerBin implies local.bin = offerTick
-        // no need to check for prevId/nextId == 0: if offer is last of leaf, it will be checked by leaf.isEmpty()
```
Comment (espendk) [inline src/MgvHasOffers.sol:79]: Should the comment be retained? I see it's a bit outdated, but still nice to have the explanation.

**Review by espendk — APPROVED** [review]
Looks good to me! Added a few questions

File: src/MgvHasOffers.sol (hunk 1 above)
Comment (adhusson) [reply src/MgvHasOffers.sol:98]: in this case, we are deleting the first offer of a bin, so the best bin might have become empty. In that case we need to execute the following if branch at the end of the `dislodgeOffer` function:

```
      if (shouldUpdateBranch) {
        local = local.binPosInLeaf(leaf.bestNonEmptyBinPos());
      }
```

File: src/MgvHasOffers.sol (hunk 2 above)
Comment (adhusson) [reply src/MgvHasOffers.sol:79]: It's being replaced by a comment in another branch :)

File: src/MgvHasOffers.sol (hunk 1 above)
Comment (espendk) [reply src/MgvHasOffers.sol:98]: ah, so it handles the case where we're deleting the best offer?

Comment (adhusson) [reply src/MgvHasOffers.sol:98]: yes!

**Pull request merged**